### ISSUE TITLE
No-Cache for entire Control Panel

### ISSUE THREAD

**Issue by lindseydiloreto** [craftcms/cms#7631]
### TL;DR

Shouldn't Craft be calling `setNoCacheHeaders` for each page of the control panel?

 - https://github.com/craftcms/cms/blob/develop/src/web/Response.php#L94

When the entire control panel is cached, we are unable to login.

---

### Full Story

Our client's host SiteGround recently made an unexpected change to their cache policy. Suddenly, the entire site was being aggressively cached, **including the control panel**. We were unable to login, **locked out of the CP entirely**.

Fortunately, someone on Discord had mentioned this issue and we were quickly able to get to the bottom of it.

 - [SiteGround Dynamic Caching - Configuration, Headers, and Purging](https://www.siteground.com/kb/siteground-dynamic-caching-configuration/)

In the support article, they mention that setting a `cache-control: no-cache` header would prevent those pages from being cached.

It makes sense that we (the website developers) would be responsible for busting that cache on the front-end. But wouldn't Craft itself be responsible for busting that cache in the control panel?

Let me know your thoughts. Curious to hear if there's a big-picture view that I'm missing.

Thanks! 🙂

**Comment by lindseydiloreto** [comment]
Here's the related conversation on Discord...

https://discord.com/channels/456442477667418113/456834258628247564/815249897095299114

**Comment by brandonkelly** [comment]
Haven’t had the need to do it because typically that’s the default for PHP requests. Craft will explicitly send no-cache headers for all CP requests.

**Comment by lindseydiloreto** [comment]
Excellent, thanks @brandonkelly! 🍺

**Comment by brandonkelly** [comment]
Craft 3.6.8 is out now with that change.
